src/core/ext/filters/client_channel/lb_policy/priority/priority.cc

```diff
@@ -380,6 +380,8 @@ void PriorityLb::HandleChildConnectivityStateChangeLocked(
         /*report_connecting=*/child_priority == current_priority_);
     return;
   }
+  // Ignore lower-than-current priorities.
+  if (child_priority > current_priority_) return;
   // The update is for a higher-than-current priority (or for any
   // priority if we don't have any current priority).
   if (child_priority < current_priority_) {
```

Comment:
This is not entered if current_prioirty_ is not unset right away in TryNextPrioirtyLocked()